client/src/lib.rs

```diff
@@ -0,0 +1,177 @@
+use ed25519_dalek::{Keypair, PublicKey};
+use failure::{err_msg, Error};
+use hpos_state_core::admin_keypair_from;
+use serde::*;
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub struct HpAdminKeypair(Keypair);
+
+#[derive(Serialize, Deserialize, Debug)]
+struct Payload {
+    method: String,
+    uri: String,
+    body_string: String,
+}
+
+#[wasm_bindgen]
+impl HpAdminKeypair {
+    /// @description Create an Ed25519 keypair for API signing calls to HPOS
+    /// @example
+    /// myKeys = new HpAdminKeypair( hc_public_key_string, email, password );
+    #[wasm_bindgen(constructor)]
+    pub fn new(
+        hc_public_key_string: String,
+        email: String,
+        password: String,
+    ) -> Result<HpAdminKeypair, JsValue> {
+        // rewrite any Error into JsValue
+        match new_inner(hc_public_key_string, email, password) {
+            Ok(v) => Ok(Self(v)),
+            Err(e) => Err(e.to_string().into()),
+        }
+    }
+
+    /// @description Sign payload and return base64 encoded signature.
+    /// Requires properly formatted payload:
+    /// const payload = {
+    ///     method: String,
+    ///     uri: String,
+    ///     body_string: String
+    /// }
+    /// @example
+    /// myKeys = new HpAdminKeypair( hc_public_key_string, email, password );
+    /// const payload = {
+    ///     method: "get",
+    ///     uri: "/someuri",
+    ///     body_string: ""
+    /// }
+    /// myKeys.sign( payload );
+    #[wasm_bindgen]
+    pub fn sign(&self, payload: &JsValue) -> Result<String, JsValue> {
+        // return meaningful error message via JsValue
+        let payload_vec = match parse_payload(payload) {
+            Ok(v) => v,
+            Err(_) => {
+                return Err("Malformed signing payload, check docs for details."
```

Comment:
Errors that are ignored here produce much better/more detailed error messages than this.  recommend to just return internal Serde error instead of using custom handling/text.